extensions/contract-termination/src/test/resources/sql/AgreementTerminationDetailsQueryTest/init.sql

```diff
@@ -0,0 +1,223 @@
+        }
+    }');
+
+-- second agreement ID's negotiation
+
+INSERT INTO public.edc_contract_negotiation (id, correlation_id, counterparty_id, counterparty_address, protocol, state,
+                                             state_count, state_timestamp, error_detail, agreement_id, trace_context,
+                                             lease_id, created_at, updated_at, type, callback_addresses, pending,
+                                             contract_offers)
+VALUES ('aed211a6-498b-4ca7-9b24-c009e7268b08', 'aed211a6-498b-4ca7-9b24-c009e7268b08', 'my-edc',
+        'http://edc:11003/api/dsp', 'dataspace-protocol-http', 1200, 1, 1717770258445, null,
+        'Y29udHJhY3Q=:YXNzZXQtMS4yLjM=:NWM4M2MzNTYtZGVlYi00NjFkLTg1ZTUtODQ0YzgwMGEwMmVm', '{}', null, 1717770255259,
+        1717770258445, 'CONSUMER', '[]', false, '[
+        {
+            "id": "Y29udHJhY3Q=:YXNzZXQtMS4yLjM=:NGJlNmI3NzYtZDdiNS00MGUwLWE4ZjktZDhkODg2Y2Q2ZDhl",
+            "policy": {
+                "permissions": [
+                    {
+                        "edctype": "dataspaceconnector:permission",
+                        "target": "asset-1.2.3",
+                        "action": {
+                            "type": "USE",
+                            "includedIn": null,
+                            "constraint": null
+                        },
+                        "assignee": null,
+                        "assigner": null,
+                        "constraints": [
+                            {
+                                "edctype": "AtomicConstraint",
+                                "leftExpression": {
+                                    "edctype": "dataspaceconnector:literalexpression",
+                                    "value": "ALWAYS_TRUE"
+                                },
+                                "rightExpression": {
+                                    "edctype": "dataspaceconnector:literalexpression",
+                                    "value": "true"
+                                },
+                                "operator": "EQ"
+                            }
+                        ],
+                        "duties": []
+                    }
+                ],
+                "prohibitions": [],
+                "obligations": [],
+                "extensibleProperties": {},
+                "inheritsFrom": null,
+                "assigner": null,
+                "assignee": null,
+                "target": "asset-1.2.3",
+                "@type": {
+                    "@policytype": "set"
+                }
+            },
+            "assetId": "asset-1.2.3"
+        }
+    ]');
+
+insert into public.sovity_contract_termination(contract_agreement_id, reason, detail, terminated_at, terminated_by)
+values ('Y29udHJhY3Q=:YXNzZXQtMS4yLjM=:NWM4M2MzNTYtZGVlYi00NjFkLTg1ZTUtODQ0YzgwMGEwMmVm', 'User Termination',
+        'Terminated because of good reasons', '2024-07-03T16:59:01.518000+00:00', 'SELF');
```

Comment:
rather having very intentful test data as code with Policy JSONs and Asset properties being "dummies" to a degree it would make sense, or you accepting `Consumer<ContractAgreementRecord>` parameters for customizing the test data between test cases